section1/reflection.md

```diff
@@ -2,18 +2,45 @@
 
+A variable is a way to define something.  It can be a word or a brief phrase, and you can decide to assign certain values to it and change those values at will by re-assigning it.
+
 6. Think of a site or app you use frequently. What are three variables that are probably used? Which data type would each of those variables probably hold?
 
+- On Amazon, variables are likely used to determine pricing.  There is usually a quantity select box which acts as a variable.  The user assigns this variable to `1`, `2` , or `3` etc. and then that variable is multiplied by the price of one unit to output the total price in the shopping cart.  In this example the data type would be a **number**.
+- Also on Amazon, a user's email and password would be variables, saved as **string** data.  The user inputs their information and if the variables entered match the variables that amazon has on file in their system, the user is able to log on.  This result of this comparison would be an example of a **boolean** data type.
+- Another example of variables on Amazon's website would be a user's address that they want something delivered to. The user inputs their address to Amazon and their system will record it as **string** data and then Amazon will know what address to send the package to.
+
 7. In your own words, explain what concatenation is.
 
+Concatenation is a way to merge variables together to form a more complex string.  The variables can be different data types, such as a number and a string, but the resulting data will be a string.
+
 8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
 
+Also on the Amazon website, when you are logged into your account it will display your name and your address near the searchbar and read something like "Deliver to 'Thomas' at 'Broomfield 80021'" This is likely an example of concatenation where they are taking the variables `userName` and `userZipCode` and putting them together to form a string that shows the user that they can deliver to their specific account/zipcode combination.
+
 9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+
+I feel like I am starting to understand how these concepts are used in real world scenarios, like with the above examples on the Amazon website, but I also feel like I'm missing other obvious examples.  I feel like Amazon probably uses variables and concatenation and interpolation all over their website, and it took me some serious thinking to just come up with the few examples above.  So I guess my question would be how can I apply these concepts to more real-world situations, rather than just typing simple sentences like we did in the exercises?  It seems like for the exercises it would be so much simpler to just type out the sentence as a string rather than assign variables and insert those variables into the sentence, so why would we do it this way?
```

Comment:
Yes, I completely understand where you are coming from.

Let's think about a product on Amazon. Each product displayed on the page follows a format, right? We see a picture, price, title, maybe the rating, and possible a couple other pieces of data. Then we see dozens of other products on the same page, that follow that exact same template. EACH of those templates used the exact same variables - the developers only declared each of those variables one time. So one might be `productName`, `productPrice`, `productRating` etc. - and we see dozens of them on the same page. 

IMO - this will continue to feel a bit contrived until you start pulling in data from an API, or back-end. But, you will start building front-end applications in the first few weeks of Mod 1 which will start bringing it to life, to an extent. I hope that Amazon example helped a bit though.